src/entity/Entity.php

```diff
@@ -785,17 +786,34 @@ protected function broadcastMovement(bool $teleport = false) : void{
 				$this->spawnTo($player);
 			}
 		}else{
-			$this->server->broadcastPackets($this->hasSpawned, [MoveActorAbsolutePacket::create(
-				$this->id,
-				$this->getOffsetPosition($this->location),
-				$this->location->pitch,
-				$this->location->yaw,
-				$this->location->yaw,
-				(
-					//TODO: if the above hack for #4394 gets removed, we should be setting FLAG_TELEPORT here
-					($this->onGround ? MoveActorAbsolutePacket::FLAG_GROUND : 0)
-				)
-			)]);
+			$deltaLoc = $offsetPosition->subtractVector($this->getOffsetPosition($this->lastLocation));
```

Comment:
This calculation is useless. With the current code, you can directly compare lastLocation->whatever with currentLocation->whatever.